include/conf/load.hpp

```diff
@@ -37,17 +38,60 @@ namespace color_coded
       }
     }
 
-    inline args_t load(std::string const &file, std::string const &filetype)
+    inline args_t load_compilation_database(std::string const &file, fs::path filename)
     {
-      if(file.empty())
-      { return defaults(filetype); }
+      static const std::string source_extensions[] {".c", ".cpp", ".cc"};
+      static const std::string header_extensions[] {".h", ".hpp", ".hh"};
+      std::string error;
+      auto const database_ptr(::clang::tooling::JSONCompilationDatabase::loadFromFile(file, error));
+      if(!database_ptr)
+      { return {}; }
+
+      std::vector<fs::path> files{filename};
+      auto const ext(filename.extension());
+      if(std::find(begin(header_extensions), end(header_extensions), ext) != end(header_extensions))
+      {
+        auto path = filename.string();
+        auto const include_it = path.rfind("include");
+        if(include_it != std::string::npos)
+        {
+            filename = path.replace(include_it, 7, "src");
```

Comment:
What do you think about `boost::range::replace(std::begin(path), "include/", "src/");` here? We're limited to boost::range, instead of boost's string algorithms, since bringing in more of boost would be a hassle; still, it's generic and iterator-based, and we're only looking for ascii. It'd allow us to get rid of the magic 7 and the whole if statement.